backend/app/classification/cache_helpers.py

```
@@ -0,0 +1,222 @@
+    future = loop.create_future()
+    prev = _inflight.setdefault(key, future)
+    if prev is not future:
+        # another caller is in-flight; await its result/failure
+        return await prev
+
+    # we are the owner; perform the fetch and set the future result/exception
+    async def _owner_fetch():
+        try:
+            start = time.time()
+            response = await llm.ainvoke([HumanMessage(content=prompt)])
+            elapsed = time.time() - start
```

Ruff (0.14.11)

Local variable `elapsed` is assigned to but never used

Remove assignment to unused variable `elapsed`

(F841)